### PR TITLE
Fix ECC Error Handling

### PR DESCRIPTION
**Pull request by Simon-Cochrane**
Rather than raising all hell and panicking when ECC errors are detected, well maybe just erase the errors.

### CODE REVIEW

File: cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #define LOG_MODULE_NAME mcumgr_flash_mgmt
 #define LOG_LEVEL CONFIG_IMG_MANAGER_LOG_LEVEL
 #include <logging/log.h>
-LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+LOG_MODULE_REGISTER(zephyr_img_mgmt);
```
Comment (alsrgv) [inline cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:23]: Change LOG_MODULE_NAME instead?

File: cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c [hunk 2]
```diff
@@ -73,7 +73,8 @@ zephyr_img_mgmt_flash_check_empty(uint8_t fa_id, bool *out_empty)
         rc = flash_area_read(fa, addr, data, bytes_to_read);
         if (rc != 0) {
             flash_area_close(fa);
-            return MGMT_ERR_EUNKNOWN;
+            LOG_ERR("Read Error, Might be Corrupted");
+            return MGMT_ERR_ECORRUPT;
```
Comment (alsrgv) [inline cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:77]: Is there a retcode for ECC error that could be detected?

**Review by alsrgv — APPROVED** [review]

File: cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c (hunk 1 above)
Comment (Simon-Cochrane) [reply cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:23]: It doesn't work :) It prints out as LOG_MODULE_NAME

File: cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c (hunk 2 above)
Comment (Simon-Cochrane) [reply cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:77]: There is not :/

File: cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c (hunk 1 above)
Comment (alsrgv) [reply cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:23]: Oh!

So perhaps nuke first two lines, and just do `LOG_MODULE_REGISTER(zephyr_img_mgmt, CONFIG_IMG_MANAGER_LOG_LEVEL);` ?

**Pull request merged**